app/models/tenant.rb

```diff
@@ -317,6 +318,20 @@ def create_miq_product_features_for_tenant_nodes
     MiqProductFeature.seed_single_tenant_miq_product_features(self)
   end
 
+  def destroy_child_tenants
+    all_subtenants.each do |subtenant|
+      subtenant.destroy_child_tenants if subtenant.all_subtenants.count > 0
```

Comment:
The conditional is not necessary, as calling this method on the subtenant will loop over nothing, which is effectively checking the number of records anyway.

Reply:
right, it is matter of preference: invoke method even it is going to execute 0 iterations or check if method need to be invoked